config/stp.py

```diff
@@ -331,45 +505,92 @@ def spanning_tree(_db):
 ###############################################
 
 # cmd: STP enable
+# Modifies & sets parameters in different tables for MST & PVST
+# config spanning_tree enable <pvst|mst>
 @spanning_tree.command('enable')
-@click.argument('mode', metavar='<pvst>', required=True, type=click.Choice(["pvst"]))
+@click.argument('mode', metavar='<pvst|mst>', required=True, type=click.Choice(["pvst","mst"]))
 @clicommon.pass_db
 def spanning_tree_enable(_db, mode):
     """enable STP """
     ctx = click.get_current_context()
     db = _db.cfgdb
-    if mode == "pvst" and get_global_stp_mode(db) == "pvst":
+    current_mode = get_global_stp_mode(db)
+
+    if mode == "pvst" and current_mode == "pvst":
         ctx.fail("PVST is already configured")
-    fvs = {'mode': mode,
-           'rootguard_timeout': STP_DEFAULT_ROOT_GUARD_TIMEOUT,
-           'forward_delay': STP_DEFAULT_FORWARD_DELAY,
-           'hello_time': STP_DEFAULT_HELLO_INTERVAL,
-           'max_age': STP_DEFAULT_MAX_AGE,
-           'priority': STP_DEFAULT_BRIDGE_PRIORITY
-           }
-    db.set_entry('STP', "GLOBAL", fvs)
-    # Enable STP for VLAN by default
-    enable_stp_for_interfaces(db)
-    enable_stp_for_vlans(db)
+    elif mode == "mst" and current_mode == "mst":
+        ctx.fail("MST is already configured")
+
+    if mode == "pvst":
+        disable_global_mst(db)
+
+        fvs = {'mode': mode,
+            'rootguard_timeout': STP_DEFAULT_ROOT_GUARD_TIMEOUT,
+            'forward_delay': STP_DEFAULT_FORWARD_DELAY,
+            'hello_time': STP_DEFAULT_HELLO_INTERVAL,
+            'max_age': STP_DEFAULT_MAX_AGE,
+            'priority': STP_DEFAULT_BRIDGE_PRIORITY
+            }
+        db.set_entry('STP', "GLOBAL", fvs)
+        
+        enable_stp_for_interfaces(db)
+        enable_stp_for_vlans(db) # Enable STP for VLAN by default
+    
+    elif mode == "mst":
+        disable_global_pvst(db)
+
+        fvs = {'mode': mode,
+               }
+        db.mod_entry('STP', "GLOBAL", fvs)
+
+        bridge_mac = get_bridge_mac_address(db)
+        if not bridge_mac:
+            ctx.fail("Bridge MAC address not found in DEVICE_METADATA table")
+
+        # Setting MSTP parameters in the STP_MST_TABLE
+        mst_fvs = {
+            'name': bridge_mac,
+            'revision': MST_DEFAULT_REVISION,
+            'max_hop': MST_DEFAULT_HOPS,
+            'max_age': MST_DEFAULT_MAX_AGE,
+            'hello_time': MST_DEFAULT_HELLO_INTERVAL,
+            'forward_delay': MST_DEFAULT_FORWARD_DELAY
+        }
+        db.set_entry('STP_MST', "GLOBAL", mst_fvs)
+
+        do_vlan_to_instance0(db) # VLANs to Instance 0 mapping as part of global configuration
```

Comment:
do_vlan_to_instance0 - this call is also not required. When MST mode is enabled , STPd will add all VLANs to CIST

Reply:
I have addressed this comment, please review